allennlp/common/file_utils.py

```diff
@@ -60,7 +61,7 @@ def filename_to_url(filename: str, cache_dir: str = None) -> Tuple[str, str]:
 
     return url, etag
 
-def cached_path(url_or_filename: str, cache_dir: str = None) -> str:
```

Comment:
@joelgrus does this `Union`, and all the `str(path)` calls in this PR change how you feel about `Path`?